### PR TITLE
Add sorted markers tree

### PR DESCRIPTION
**Pull request by mharis001**
**When merged this pull request will:**
- Replace the default zeus markers tree with a custom one with markers sorted by category (like 3DEN)

<details>
<summary>Image</summary>

markers_tree

</details>

### CODE REVIEW

File: addons/markers_tree/functions/fnc_compile.sqf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * Arguments:
+ * None
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [] call zen_markers_tree_fnc_compile
+ *
+ * Public: None
```
Comment (CreepPork) [inline addons/markers_tree/functions/fnc_compile.sqf]:
```suggestion
 * Public: No
```

File: addons/editor/functions/fnc_handleTreeButtons.sqf [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 params ["_expand"];
 
-// Cant collapse marker or recent trees
+// Cant collapse or expand marker or recent trees
```
Comment (CreepPork) [inline addons/editor/functions/fnc_handleTreeButtons.sqf]:
```suggestion
// Can't collapse or expand marker or recent trees
```

File: addons/markers_tree/functions/fnc_handleCustomSelect.sqf [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [CONTROL] call zen_markers_tree_fnc_handleCustomSelect
+ *
+ * Public: No
+ */
+
+params ["_ctrlTreeCustom", "_selectedPath"];
```
Comment (CreepPork) [inline addons/markers_tree/functions/fnc_handleCustomSelect.sqf]: Params don't match docblock

File: addons/markers_tree/functions/fnc_handleEngineSelect.sqf [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [CONTROL] call zen_markers_tree_fnc_handleEngineSelect
+ *
+ * Public: No
+ */
+
+params ["_ctrlTreeEngine", "_selectedPath"];
```
Comment (CreepPork) [inline addons/markers_tree/functions/fnc_handleEngineSelect.sqf]: Params don't match dockblock

**Comment by oOKexOo** [comment]
Maybe make the component a bit more general by naming it `markers` instead.

**Comment by mharis001** [comment]
I disagree, this component should only implement the custom markers tree and the name `markers_tree` makes that clear. If any other changes or additions relating to markers are made in the future, they shouldn't be placed in this component.

File: addons/editor/script_idc.hpp [hunk 5]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (CreepPork) [inline addons/editor/script_idc.hpp]: What's with this blank line?

**Pull request merged**